### PR TITLE
add io.github.NickKarpowicz.LightwaveExplorer

### PR DESCRIPTION
**Pull request by NickKarpowicz**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions.
    - Extra permission requested is --filesystem=home. This is due to the nature of how the simulation code should run: The user enters the path where all of the results should be saved and can leave a queue that runs for a long time (some students have had them go for a full day). In these cases it's better if the results are saved immediately and automatically without further interaction. 
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:[I am the sole developer of the project, repo here](https://github.com/NickKarpowicz/LightwaveExplorer) **
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by NickKarpowicz** [comment]
Thank you very much @hfiguiere for the improvements made on my [last PR](https://github.com/flathub/flathub/pull/4423). Since I made a number of errors in the submission process, I created a new PR with a new version of the manifest which addresses your comments:
 - I now use fallback-X11 socket
 - hashes have been changed to sha256 from md5
 - buildsystems have been changed to cmake-ninja, including that of the main repo, which used a shell script earlier

File: io.github.NickKarpowicz.LightwaveExplorer.json [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                }
+            ]
+        },
+        {
+            "name": "LightwaveExplorer",
+            "buildsystem": "cmake-ninja",
+            "config-opts": ["-DMAKECUDAFLATPAK=TRUE", "-DCMAKE_CUDA_ARCHITECTURES=75", "-DCUDAToolkit_ROOT=/app/opt/cuda","-DCMAKE_CUDA_COMPILER=/app/opt/cuda/bin/nvcc"],
+            "build-options": {
+                "cxxflags": "-O3 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -grecord-gcc-switches",
+                "cxxflags-override": true},
```
Comment (hfiguiere) [inline io.github.NickKarpowicz.LightwaveExplorer.json:134]: if the code triggers the assertion it should be fixed. disabling the assertions is just hiding the potential crashes.

File: io.github.NickKarpowicz.LightwaveExplorer.json [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                    "type": "git", 
+                    "url": "https://github.com/davisking/dlib", 
+                    "tag": "v19.24.2",
+                    "commit": "39749fba5163d4e25030358f6aa8d91617bfdca2" } 
+                ]
+        },
+        {
+            "name": "cuda",
+            "buildsystem": "simple",
+            "build-commands":["mkdir /app/tmp","sh cuda_12.2.1_535.86.10_linux.run --silent --toolkitpath=/app/opt/cuda --toolkit --override --tmpdir=/app/tmp"],
```
Comment (hfiguiere) [inline io.github.NickKarpowicz.LightwaveExplorer.json:119]: you still haven't explained how the aarch64 flatpak that is built will work (or even build)

**Comment by hfiguiere** [comment]
bot, build io.github.NickKarpowicz.LightwaveExplorer

File: io.github.NickKarpowicz.LightwaveExplorer.json [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                    "type": "file",
+                    "url": "https://developer.download.nvidia.com/compute/cuda/12.2.1/local_installers/cuda_12.2.1_535.86.10_linux.run",
+                    "sha256": "c51e814eaa756160a73289b1f35880cc38c18e1ddedc87f46f7ff0d1696bdb95"
+                }
+            ]
+        },
+        {
+            "name": "LightwaveExplorer",
+            "buildsystem": "cmake-ninja",
+            "config-opts": ["-DMAKECUDAFLATPAK=TRUE", "-DCMAKE_CUDA_ARCHITECTURES=75", "-DCUDAToolkit_ROOT=/app/opt/cuda","-DCMAKE_CUDA_COMPILER=/app/opt/cuda/bin/nvcc"],
```
Comment (hfiguiere) [inline io.github.NickKarpowicz.LightwaveExplorer.json:131]: please format the JSON properly. its hard to read,

File: io.github.NickKarpowicz.LightwaveExplorer.json (hunk 2 above)
Comment (NickKarpowicz) [reply io.github.NickKarpowicz.LightwaveExplorer.json:118]: Is there a way to disable aarch64 entirely? The other dependency OneAPI is not compatible with it.

Comment (NickKarpowicz) [reply io.github.NickKarpowicz.LightwaveExplorer.json]: I added an only-arches command to this effect, commit a46b776fa28f8f4821df3d1a02c37a681c206daf

Comment (hfiguiere) [reply io.github.NickKarpowicz.LightwaveExplorer.json]: this won't do it

Comment (hfiguiere) [reply io.github.NickKarpowicz.LightwaveExplorer.json]: https://docs.flathub.org/docs/for-app-authors/maintanance/#flathubjson

**Comment by hfiguiere** [comment]
bot, build io.github.NickKarpowicz.LightwaveExplorer

File: io.github.NickKarpowicz.LightwaveExplorer.json (hunk 1 above)
Comment (NickKarpowicz) [reply io.github.NickKarpowicz.LightwaveExplorer.json]: I modified the code and point the manifest to the new tagged version in commit 40eac10c4e2c8ed439b978be97cbfd37708f786e. Earlier, I would check if the result of a call to std::exp() was NaN, which would cause the assertion to fail for a bad argument; now I check the argument first. Simulation results are the same, but now I don't override the build options.

**Comment by NickKarpowicz** [comment]
I noticed that when launched the flatpak built by the buildbot has the default x11 icon in the KDE taskbar, while the one that I built with flatpak-builder locally has the application's icon. Is this normal behavior?

File: io.github.NickKarpowicz.LightwaveExplorer.json (hunk 2 above)
Comment (NickKarpowicz) [reply io.github.NickKarpowicz.LightwaveExplorer.json]: Build succeeds after making flathub.json; thanks

**Comment by NickKarpowicz** [comment]
bot, build io.github.NickKarpowicz.LightwaveExplorer

**Comment by NickKarpowicz** [comment]
I tested the flatpak on a few machines and I think it works well now. Is there anything else I should change in order to proceed?

**Comment by NickKarpowicz** [comment]
bot, build io.github.NickKarpowicz.LightwaveExplorer

File: io.github.NickKarpowicz.LightwaveExplorer.json [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            "sources": [
+                {
+                    "type": "file",
+                    "url": "https://registrationcenter-download.intel.com/akdlm/IRC_NAS/992857b9-624c-45de-9701-f6445d845359/l_BaseKit_p_2023.2.0.49397_offline.sh",
+                    "sha256": "3dd04d2cc9ba2a8785580ba09068dbaf9aff9aa930b5fa7f10688210ea9be4a0"
+        
+                }
+            ],
+            "cleanup":[
+                "/app/opt/intel/oneapi/mkl/latest/lib/ia32",
```
Comment (hfiguiere) [inline io.github.NickKarpowicz.LightwaveExplorer.json:78]: this doesn't do what you think it does.

>        cleanup (array of strings)
>           An array of file patterns that should be removed at the end. Patterns starting with / are taken to be full pathnames (without the /app prefix),
>           otherwise they just match the basename.

tl;dr the `/` of from the `$FLATPAK_DEST`.

Comment (NickKarpowicz) [reply io.github.NickKarpowicz.LightwaveExplorer.json]: thank you; I removed /app from the paths in commit 083be8aa04c0f8951a5521a53eab4f4c0024ac5b

**Comment by NickKarpowicz** [comment]
The patched version to work better in Flatpak form (config/database search path starts in home rather than ../share relative to binary) passed my tests.
